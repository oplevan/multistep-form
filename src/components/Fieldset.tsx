import React from "react";

interface FieldSetProps {
  legend: string;
  children: React.ReactNode;
}

export default function Fieldset({ legend, children }: FieldSetProps) {
  return (
    <fieldset
      data-testid="fieldset"
      className="border rounded-md px-3 pb-4 space-y-4"
    >
      <legend className="bg-white px-2 text-slate-700">{legend}</legend>
      {children}
    </fieldset>
  );
}
