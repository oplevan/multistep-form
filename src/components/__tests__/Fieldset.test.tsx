import { render, screen } from "@testing-library/react";
import Fieldset from "../Fieldset";

describe("<Fieldset /> component", () => {
  test("renders correctly", () => {
    const { container } = render(
      <Fieldset legend="TEST LEGEND">
        <div>Test Child</div>
      </Fieldset>
    );
    expect(container).toMatchSnapshot();
  });

  test("renders legend correctly", () => {
    render(<Fieldset legend="Test Legend">TEST</Fieldset>);
    const legendElement = screen.getByText("Test Legend");
    expect(legendElement).toBeInTheDocument();
  });

  test("renders children correctly", () => {
    render(
      <Fieldset legend="Test Legend">
        <div data-testid="test-child">Test Child</div>
      </Fieldset>
    );
    const childElement = screen.getByTestId("test-child");
    expect(childElement).toBeInTheDocument();
  });

  test("applies correct class names", () => {
    render(<Fieldset legend="Test Legend">TEST</Fieldset>);
    const fieldsetElement = screen.getByTestId("fieldset");
    expect(fieldsetElement).toHaveClass("border");
    expect(fieldsetElement).toHaveClass("rounded-md");
    expect(fieldsetElement).toHaveClass("px-3");
    expect(fieldsetElement).toHaveClass("pb-4");
    expect(fieldsetElement).toHaveClass("space-y-4");
  });
});
