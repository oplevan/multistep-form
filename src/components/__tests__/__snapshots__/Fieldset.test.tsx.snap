// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Fieldset /> component renders correctly 1`] = `
<div>
  <fieldset
    class="border rounded-md px-3 pb-4 space-y-4"
    data-testid="fieldset"
  >
    <legend
      class="bg-white px-2 text-slate-700"
    >
      TEST LEGEND
    </legend>
    <div>
      Test Child
    </div>
  </fieldset>
</div>
`;
